src/tools/wasm-opt.cpp

```diff
@@ -359,7 +363,11 @@ int main(int argc, const char* argv[]) {
     auto runPasses = [&]() {
       options.runPasses(wasm);
       if (options.passOptions.validate) {
-        bool valid = WasmValidator().validate(wasm);
+        WasmValidator::Flags flags = WasmValidator::Globally;
+        if (options.passOptions.closedWorld) {
+          flags |= WasmValidator::ClosedWorld;
```

Comment:
This pattern repeats a bit - I wonder if the `WasmValidator` constructor could perhaps take as a parameter a `PassOptions` and read what it needs from there? But anyhow that could be a followup.